src/lib/MasonryLayout/index.tsx

```diff
@@ -0,0 +1,43 @@
+import { ElementType, ReactElement, createElement, forwardRef } from 'react';
+import { PolymorphicComponentProps, PolymorphicRef } from '../types/common';
+import * as S from './style';
+import MasonryLayoutItem from './MasonryLayoutItem';
+
+export interface _MasonryLayoutProps {
+  itemSize: string;
+  gap?: string;
+  rowGap?: string;
+  columnGap?: string;
+}
+
+export type MasonryLayoutProps<T extends ElementType> =
+  PolymorphicComponentProps<T, _MasonryLayoutProps>;
+
+type MasonryLayoutComponent = <T extends ElementType>(
+  props: MasonryLayoutProps<T>
+) => ReactElement | null;
+
+const MasonryLayout: MasonryLayoutComponent = forwardRef(function MasonryLayout<
+  T extends ElementType = 'div'
+>(
+  { css, as, children, ...rest }: MasonryLayoutProps<T>,
+  ref: PolymorphicRef<T>
+) {
+  const childArray = Array.isArray(children) ? children : [children];
+
+  return (
+    <S.Component as={as} ref={ref} style={css} {...rest}>
+      {childArray.map((child) => (
+        <MasonryLayoutItem rowGap={rest.rowGap} gap={rest.gap}>
+          {createElement(child.type, {
+            ...{
+              ...child.props,
+            },
```

Comment:
```suggestion
              ...child.props,
```
위와 같이 변경할 수 도 있지 않을까요?

Reply:
디테일하게 봐주셔서 감사합니다 👍 

클린이 제안해준 코드로 고쳐볼 수 있을 것 같아요 💯